pkg/buildpipeline/pipeline.go

```diff
@@ -797,20 +799,28 @@ func findWorkspaceProvider(block TransformedStage, blocks []TransformedStage) (b
 		return true, nil
 	}
 
+	var parallelEndTasks []string
+
 	for i := len(blocks) - 1; i >= 0; i-- {
 		previousBlock := blocks[i]
 		if len(previousBlock.Sequential) > 0 {
 			found, provider := findWorkspaceProvider(block, previousBlock.Sequential)
 			if found {
-				return true, provider
+				return true, append(parallelEndTasks, provider...)
 			}
-		} else if len(previousBlock.Sequential) > 0 {
+		} else if len(previousBlock.Parallel) > 0 {
 			// We don't want to use a workspace from a parallel block outside of that block.
 			// TODO: What to do about custom workspaces? Check for erroneous uses specially?
 			// Allow them if only one of the parallel tasks uses the same resource?
+
+			if previousBlock.Depth == block.Depth {
+				// TODO: We do want to record the last task in each parallel branch, though, currently hacked in as from,
+				// but will end up being used for the non-resource-based dependency ordering field.
```

Comment:
Ah gotcha, we still need to make sure that our input ends up being executed block-by-block even if there are no specific resource-level dependencies.

Reply:
Yeah, lexical order is still A Thing. =)